### DISCUSSION TITLE
How can I get seed phrase with react native sdk?

### DISCUSSION THREAD

**Discussion by ruymaster** [Web3Auth/Web3Auth#821] [Help]
I am trying to integrate web3auth in our wallet app. Wallet app is made in react native. I need to get seed phrase for each user, so users can login with web3auth  and create wallet automatically. How can I get seed phrase with react native sdk?
I used custom verifier. My code is like this.
const web3Auth = new Web3Auth(WebBrowser, {
			clientId: 'xxxxx',
			network: OPENLOGIN_NETWORK.TESTNET,
			loginConfig: {
				google: {
					verifier: 'my_verifier', 
					typeOfLogin: 'google',
					name: 'Custom Login with Google',
					clientId: 'my_google_client_id'
				}
			}
		});
		const state = await web3Auth.login({ loginProvider: LOGIN_PROVIDER.GOOGLE, redirectUrl: resolvedRedirectUrl });

**Comment by shahbaz17** [comment]
Hey @ruymaster 

> "I need to get seed phrase for each user, ..."
>> Web3Auth returns user's private key upon user's successful authentication. And that can be used to create a provider and use it to make blockchain calls.

Please look at any react native apps present at https://github.com/Web3Auth/examples/tree/main/react-native and try running it and see if this is what you need.

**Comment by ruymaster** [comment]
Thanks for your response. I already made some modules based on example. So I got private key with web3auth. But I can't get 12 seed phrase or store generated randomly seed phrase string, as simular as binance wallet extension doing.

**Reply by yashovardhan** [reply]
You need to use a bip 32 package to convert private key to seed phrases. Have a look at this one: https://www.npmjs.com/package/react-native-bip32-utils

**Comment by ruymaster** [comment]
Thanks for replying. Private key can't be converted to seed phrases. You can check this link.
https://ethereum.stackexchange.com/questions/62530/how-can-i-get-a-mnemonic-phrase-from-a-private-key

**Reply by yashovardhan** [reply]
Hey @ruymaster, the answer says, **a mnemonic is generated using an hd wallet**. 

You can use the web3auth private key as a BIP39 seed root key for that. Checkout this website, where you can customise this: https://iancoleman.io/bip39/

**Comment by ruymaster** [comment]
Thanks, @yashovardhan I understand. It is possible to generate seed with enough entropy. thanks again.
